### ISSUE TITLE
Big Sur says you don't have permission to run DOSBox-X. If I lay out candles in Mac SE formation and pray to the Holy Spirit of Steve Jobs might I appease the Apple gods? FFS

### ISSUE THREAD

**Issue by joncampbell123** [joncampbell123/dosbox-x#2152]
**Describe the bug**
Big Sur tells users they don't have permission to run the application they just downloaded.

**To Reproduce**
1. Download the Mac binaries from releases
2. Unpack ZIP
3. Try to run

**Expected behavior**
Gatekeepers warns the user the app is unsigned, allows them to run it anyway.

See also:
https://mobile.twitter.com/jockking/status/1345861727653683200

**Comment by Skaggynuts** [comment]
DosBox isn't executable.
Right Click Dosbox.
Show Package contents.
Open Contents
Open MacOS
Drag DosBox from folder to desktop
Make Exectable by running: _chmod 700 dosbox_ from terminal.

**Comment by joncampbell123** [comment]
Is Finder unpacking the ZIP file and NOT preserving Unix permissions?

I just double checked both Mac OS ZIP archives and the permissions of the DosBox executable are 0755 here, but maybe that's just InfoZIP doing it's thing by default.

Can you unzip the ZIP file on Mac OS and confirm what file permissions are assigned after unpacking?

Maybe someone at Apple thought they could protect their dumb users by removing the -x attribute when extracting from ZIP archives. Maybe it's time for DOSBox-X to use DMG images.

**Comment by Skaggynuts** [comment]
664

**Comment by Wengier** [comment]
@Skaggynuts By the newest version, do you mean the notarized version? It is also available here:

* https://github.com/joncampbell123/dosbox-x/releases/download/dosbox-x-v0.83.9/dosbox-x-macosx-x86_64-signed.zip

**Comment by joncampbell123** [comment]
> 664 with Archive Utility (Apple)
> 755 if I extract with the Unarchiver (3rd Party)
> ಠ_ಠ
> 
> *That was with the version I downloaded on January 1st. The newest download runs fine.

Damn it Apple, did you guys forget all the third party software already out there distributed in ZIP format?

*sigh* :neutral_face: 

Does this explain the other complaint where Big Sur claims DOSBox-X is damaged and won't let you run it?

**Comment by joncampbell123** [comment]
So then, next question: What do we ship DOSBox-X in if ZIP archives are no longer a valid way to do it?

Is there some command line magic in OS X to pack an .app into a DMG image?

**Comment by Skaggynuts** [comment]
1. Seems the problem resolved its self once the package was notarized and I re- downloaded it, but if you want use use the DMG route it’s fairly easy to create a DMG package from the disk utility app.

2. The complaint about Dosbox being damaged only applies to the ARM version, as far as I can tell its impossible to install any unsigned ARM apps without disabling both Gatekeeper and SIP.

**Comment by emendelson** [comment]
> Damn it Apple, did you guys forget all the third party software already out there distributed in ZIP format?
> 
> I guess that means no more distributing binaries in ZIP archives.

If you use a macOS utility to ZIP the files, the extended attributes will be preserved on unzipping on a Mac, as far as I can see.

**Comment by joncampbell123** [comment]
My makefile uses the command line zip tool to make the ZIP archive I upload. Is that the same?

**Comment by emendelson** [comment]
Yes, that should work, I believe. No one else has reported a problem with extended attributes, have they?

**Comment by brunocastello** [comment]
This is how I work around the issue since Mojave:

> sudo spctl --master-disable
